src/GetPdoExpressionHashMapCapableTrait.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace RebelCode\Storage\Resource\Pdo;
+
+use Dhii\Expression\ExpressionInterface;
+use Dhii\Expression\LiteralTermInterface;
+use Dhii\Expression\TermInterface;
+use Dhii\Util\String\StringableInterface as Stringable;
+use InvalidArgumentException;
+
+/**
+ * Common functionality for objects that can generate an expression value hash map for use in PDO parameter binding.
+ *
+ * @since [*next-version*]
+ */
+trait GetPdoExpressionHashMapCapableTrait
+{
+    /**
+     * Retrieves the expression value hash map for a given SQL condition, for use in PDO parameter binding.
+     *
+     * @since [*next-version*]
+     *
+     * @param TermInterface         $condition    The condition instance.
+     * @param string[]|Stringable[] $ignore       A list of term names to ignore, typically column names.
+     * @param array                 $valueHashMap The value hash map reference to write to.
+     *
+     * @return array A map of value names to their respective hashes.
+     */
+    protected function _getPdoExpressionHashMap(TermInterface $condition, array $ignore = [], array &$valueHashMap = [])
+    {
+        if ($valueHashMap === null) {
```

Comment:
How about defaulting this to `null`, and not populating it in that case?

Reply:
More importantly, why both accept it by reference, and return it?

Reply:
It's accepted by reference for performance reasons. Since it's recursive, the resulting arrays would have to be merged. By passing the result by reference, no merging is needed.

But the parameter is also optional. The caller need not pass an existing map to get a resulting map. So the result is returned as well.

It might seem redundant since the caller can already retrieve the result from the argument passed by reference, but since this method had no other viable return value, I decided it just makes sense to return the map - because it _is_ the result after all.

Not populating and returning a pdo expression hash map in a method literally called `_getPdoExpressionHashMap` doesn't make sense to me.

Reply:
Re-naming to something like `_populatePdoExpressionHashMap()` could solve the problem.

Reply:
And what _is_ the problem?